codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -565,11 +720,18 @@ int incrementStreamWindow(int delta) throws Http2Exception {
             if (streamableDelta != 0) {
                 incrementStreamableBytesForTree(streamableDelta);
             }
+
+            // Check if a writability change has occurred, and save it for processing at some later point.
+            if (!writabilityChanged) {
+                writabilityChanged = previouslyWritable ^ (writableWindow() > 0);
```

Comment:
I'm not sure, do you have a specific case in mind? Seems like we only need to worry about when the allocated amount (just temporary to ensure allocation is correct / fair) is actually decremented from the window. The `writeAllocatedBytes` will eventually decrement the window (`writeAllocatedBytes` -> `writeBytes` -> `write` -> `decrementFlowControlWindow` -> `incrementStreamWindow`).